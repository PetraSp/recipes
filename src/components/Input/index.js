import React from 'react';
import PropTypes from 'prop-types';
import StyledInputContainer, { StyledInputBox } from './styles';

function Input(props) {
  const { containerWidth, inputWidth } = props;

  return (
    <StyledInputContainer containerWidth={containerWidth}>
      <StyledInputBox {...props} inputWidth={inputWidth} />
    </StyledInputContainer>
  );
}

Input.propTypes = {
  containerWidth: PropTypes.string,
  inputWidth: PropTypes.string
};

Input.defaultProps = {
  containerWidth: 'auto',
  inputWidth: 'auto'
};
export default Input;
