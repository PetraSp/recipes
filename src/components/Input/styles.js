import Styled from 'styled-components';
import COLOR_PALETTE, { FONT_FAMILY } from '../../Constants';

const StyledInputContainer = Styled.div`
    display: inline-flex;
    min-width: ${props => props.containerWidth};
`;

const StyledInputBox = Styled.input`
  min-width: ${props => props.inputWidth};
  box-shadow: 0 0 0 4px ${COLOR_PALETTE.green};
  font-family: ${FONT_FAMILY};
  flex: 1 0 auto;
  border: 0;
  height: 35px;
  font-size: 15px;
  font-weight: 500;
  display: inline-block;
  padding: 11px;
  border-radius: 3px;
  color: ${COLOR_PALETTE.black};
  &:hover {
    opacity: 0.8;
  }
  &:focus {
    outline: 0;
  }
  ::placeholder {
    color: ${COLOR_PALETTE.darkGrey};
  }
`;

export default StyledInputContainer;
export { StyledInputBox };
