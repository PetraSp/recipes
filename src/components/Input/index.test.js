import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Input from './index';

describe('Input', () => {
  it('should render', () => {
    const tree = renderer.create(<Input containerWidth="350px" inputWidth="200px" />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree.children[0]).toHaveStyleRule('height', '35px');
    expect(tree).toHaveStyleRule('min-width', '350px');
    expect(tree.children[0]).toHaveStyleRule('min-width', '200px');
  });
});
