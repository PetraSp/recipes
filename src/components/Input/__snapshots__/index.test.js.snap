// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 350px;
}

.c1 {
  min-width: 200px;
  box-shadow: 0 0 0 4px #0AEB8B;
  font-family: Poppins,sans-serif;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  border: 0;
  height: 35px;
  font-size: 15px;
  font-weight: 500;
  display: inline-block;
  padding: 11px;
  border-radius: 3px;
  color: #000000;
}

.c1:hover {
  opacity: 0.8;
}

.c1:focus {
  outline: 0;
}

.c1::-webkit-input-placeholder {
  color: #b3b3b3;
}

.c1::-moz-placeholder {
  color: #b3b3b3;
}

.c1:-ms-input-placeholder {
  color: #b3b3b3;
}

.c1::placeholder {
  color: #b3b3b3;
}

<div
  className="c0"
>
  <input
    className="c1"
  />
</div>
`;
